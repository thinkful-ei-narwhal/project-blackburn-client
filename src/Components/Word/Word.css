.word {
    text-align: center;
    padding: 10px;
    width: 12vh;
    margin: 10px;
}