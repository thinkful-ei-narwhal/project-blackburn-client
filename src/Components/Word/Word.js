import React, { Component } from "react";
import "./Word.css";

class Word extends Component {
  constructor(props) {
    super(props);
    this.state = { timer: this.props.timer / 1000 };
  }

  //errors wher rendering a countdown timer for separate word components
  // countDown(state) {
  //   if (state.timer > 0)
  //     this.setState({ timer: this.state.timer - 1 });
  // }

  // componentDidMount() {
  //   setInterval(() => this.countDown(this.state), 1000);
  // }

  // componentWillUnmount() {
  //   clearInterval(this.interval);
  // }

  render() {
    return (
      <div>
        <p className = 'word' style = {{fontSize: '24px'}}>{this.props.word}</p>
        {/* <p>{this.state.timer}</p> */}
      </div>
    );
  }
}

export default Word;
