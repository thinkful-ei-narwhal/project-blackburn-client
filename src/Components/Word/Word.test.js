import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom'
// import { PotluckProvider } from '../../context/PotluckContext'
import Word from './Word';

describe(`Adds word to dom without crashing`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(
            <Word word = {'hello'} />

    )
    .find('.word')
    expect(toJson(wrapper)).toMatchSnapshot()
    })
})