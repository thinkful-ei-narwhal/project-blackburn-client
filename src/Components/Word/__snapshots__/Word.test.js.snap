// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds word to dom without crashing renders without crashing 1`] = `
<p
  className="word"
>
  hello
</p>
`;
